tools/system_libs.py

```diff
@@ -281,6 +296,101 @@ def get_files(self):
 
     raise NotImplementedError()
 
+  def write_ninja_file(self, filename, libname):
+    cflags = self.get_cflags()
+    asflags = get_base_cflags()
+    # TODO(sbc) There is an llvm bug that causes a crash when `-g` is used with
+    # assembly files that define wasm globals.
+    asflags = [arg for arg in asflags if arg != '-g']
+    cflags_asm = [arg for arg in cflags if arg != '-g']
+
+    def join(flags):
+      return ' '.join(flags)
+
+    out = f'''\
+# Automatically generated by tools/system_libs.py.  DO NOT EDIT
+
+ninja_required_version = 1.5
+
+ASFLAGS = {join(asflags)}
+CFLAGS = {join(cflags)}
+CFLAGS_ASM = {join(cflags_asm)}
+EMCC = {shared.EMCC}
+EMXX = {shared.EMXX}
+EMAR = {shared.EMAR}
+
+rule cc
```

Comment:
If the real value is to use this for development, it seems we'll want to make sure that there are dependencies on the clang binaries, and potentially all of the python files, since changes to any of them could affect the object file output. Not all build systems do this properly, I can't recall whether Ninja is one of them.

Reply:
Oh yes.. that is unfortunately true.  I had forgotten about that.

Reply:
My guess is that that full dependency graph would be too sensitive, but that this feature it most useful for folks working the libraries code and headers within emcripten.